components/service-binding-usage-controller/cmd/migration/main.go

```diff
@@ -0,0 +1,157 @@
+	}
+
+	args := os.Args
+	manager := sbuManager{client: client, scClient: scClient}
+	switch args[1] {
+	case removeOwnerReference:
+		log.Info("Start removing owner reference from ServiceBindingUsage")
+		if err := manager.prepareServiceBindingUsages(); err != nil {
+			log.Fatalf("while prepare service binding usage to upgrade: %s", err)
+		}
+		log.Info("Removing owner reference from ServiceBindingUsage is done")
+	case addOwnerReference:
+		log.Info("Start restoring owner reference to ServiceBindingUsage")
+		if err := manager.restoreServiceBindingUsages(); err != nil {
+			log.Fatalf("while restore service binding usage after upgrade: %s", err)
+		}
+		log.Info("Restoring owner reference to ServiceBindingUsage is done")
+	default:
+		log.Fatalf("The command is not support. Available commands: %s, %s", removeOwnerReference, addOwnerReference)
+	}
+}
+
+func (sm *sbuManager) prepareServiceBindingUsages() error {
+	list, err := sm.client.ServicecatalogV1alpha1().ServiceBindingUsages(v1.NamespaceAll).List(v1.ListOptions{})
+	if err != nil {
+		return errors.Wrap(err, "while fetching ServiceBindingUsage list")
+	}
+
+	for _, sbu := range list.Items {
+		err = sm.removeOwnerReference(&sbu)
+		if err != nil {
+			return errors.Wrapf(err, "while removing owner reference from %q ServiceBindingUsage", sbu.Name)
+		}
+	}
+
+	return nil
+}
+
+func (sm *sbuManager) removeOwnerReference(sbu *v1alpha1.ServiceBindingUsage) error {
+	toUpdate := sbu.DeepCopy()
+	toUpdate.ObjectMeta.OwnerReferences = nil
+
+	_, err := sm.client.ServicecatalogV1alpha1().ServiceBindingUsages(toUpdate.Namespace).Update(toUpdate)
+	if err != nil {
+		return errors.Wrapf(err, "while updating %q ServiceBindingUsage", toUpdate.Name)
+	}
+
+	return nil
+}
+
+func (sm *sbuManager) restoreServiceBindingUsages() error {
+	listSBU, err := sm.client.ServicecatalogV1alpha1().ServiceBindingUsages(v1.NamespaceAll).List(v1.ListOptions{})
+	if err != nil {
+		return errors.Wrap(err, "")
+	}
+
+	for _, sbu := range listSBU.Items {
+		if sm.hasOwnerReference(sbu) {
+			continue
+		}
+		sb, err := sm.findServiceBinding(sbu.Spec.ServiceBindingRef.Name, sbu.Namespace)
+		if err != nil {
+			return errors.Wrapf(err, "while finding ServiceBinding %q", sbu.Spec.ServiceBindingRef.Name)
+		}
+		err = sm.updateOwnerReference(sbu, sb)
+		if err != nil {
+			return nil
+		}
+	}
+
+	return nil
+}
+
+func (sm *sbuManager) hasOwnerReference(sbu v1alpha1.ServiceBindingUsage) bool {
+	for _, or := range sbu.OwnerReferences {
+		if or.Name == sbu.Spec.ServiceBindingRef.Name {
+			return true
+		}
+	}
+
+	return false
+}
+
+func (sm *sbuManager) findServiceBinding(name, namespace string) (*v1beta1.ServiceBinding, error) {
+	if sm.sbList == nil {
+		list, err := sm.scClient.ServicecatalogV1beta1().ServiceBindings(v1.NamespaceAll).List(v1.ListOptions{})
+		if err != nil {
+			return nil, errors.Wrap(err, "while fetching ServiceBinding list")
+		}
+		sm.sbList = list
```

Comment:
tbh I'm not convinced to have such caching, we will not have too much SBU, so maybe is better to always get related ServiceBinding and have always fresh data?

IMO we can run into some issues with simple such solution, if you want to do sth like that then maybe is better to create informer which has dynamic caching so we will have always fresh data